lib/bounces.js

```diff
@@ -49,6 +49,44 @@ module.exports = function (log, error) {
       }
     }
 
+    function getHeaderValue(headerName, message){
+      var value = ''
+      if (message.mail && message.mail.headers) {
+        message.mail.headers.some(function (header) {
+          if (header.name === headerName) {
+            value = header.value
+            return true
+          }
+
+          return false
+        })
+      }
+
+      return value
+    }
+
+    function logBounceEvent(message) {
+      var verifyCode = getHeaderValue('X-Verify-Code', message)
+      var uid = getHeaderValue('X-Uid', message)
+
+      if (uid && verifyCode) {
+        var fakeRequestObject = {
+          auth: {
+            credentials: {
+              uid: uid,
+              id: verifyCode
+            }
+          },
+          headers: message.headers
+        }
+
+        return log.activityEvent('account.confirmation.bounced', fakeRequestObject, {
```

Comment:
Just piling in to say I agree with both @vladikoff's points here. It should call `log.flowEvent` directly and, in order to get the flow id out of memcached, the event name will need to be stashed first.

As mentioned in the discussion Vlad linked to though, the way that stashing business has turned out is a mistake. The sooner we get all the `metricsContext` stuff properly refactored, the better. So maybe, in light of that, the new flow event can wait until that stuff is refactored to be easier to use? Once it's done we'll probably want to go on a big flow-eventing spree anyway.